### PR TITLE
Feature/swaps filters

### PR DESCRIPTION
**Pull request by sdjuka**

### CODE REVIEW

File: src/modules/swaps/dto/swap-options.dto.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  MaxDate,
+  Min,
+  MinDate,
+} from 'class-validator';
+import { getDateOneMonthBefore } from 'src/utils/date-utils';
+
+const DEFAULT_DATE_FROM = getDateOneMonthBefore();
+const DEFAULT_DATE_TO = new Date();
+const DEFAULT_MIN_AMOUNT = 0;
+const DEFAULT_MAX_AMOUNT = 1000000000;
```
Comment (DanijelCBS) [inline src/modules/swaps/dto/swap-options.dto.ts:18]: Put max value for number here

File: src/modules/swaps/swaps.repository.ts [hunk 2]
```diff
@@ -17,16 +28,87 @@ export class SwapRepository {
...
+  async findAllSwaps(
+    pageOptions: PageOptionsDto,
+    swapOptions: SwapOptionsDto,
+  ): Promise<PageDto<SwapDto>> {
+    const queryBuilder: SelectQueryBuilder<Swap> =
+      this.swapRepository.createQueryBuilder('swap');
+
+    const whereClauses = this.getWhereClauses(swapOptions);
+
+    if (whereClauses.length > 0) {
```
Comment (DanijelCBS) [inline src/modules/swaps/swaps.repository.ts:71]: Unnecessary check

File: src/modules/swaps/swaps.repository.ts [hunk 3]
```diff
@@ -17,16 +28,87 @@ export class SwapRepository {
     private readonly swapMapper: SwapEntityToDto,
   ) {}
 
+  private getWhereClauses(swapOptions: SwapOptionsDto): WhereClause[] {
```
Comment (DanijelCBS) [inline src/modules/swaps/swaps.repository.ts:31]: Move this function to the end of file

**Review by DanijelCBS — CHANGES_REQUESTED** [review]

**Review by Str4LeGit — APPROVED** [review]
Very nice! I like the way you documented dto fields.

**Review by DanijelCBS — APPROVED** [review]

**Pull request merged**